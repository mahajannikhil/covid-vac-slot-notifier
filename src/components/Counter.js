import React, { Component } from 'react'
import glamorous from 'glamorous'
import Button from './Button'

const Wrapper = glamorous.div({}, (props, theme) => ({
  color: theme.colors.green
}))

class Counter extends Component {
  increment (text: string, e: any): void {
    // maybe I'd like to do something here
    // for the moment I just call the callback passed from the parent
    return this.props.onIncrement()
  }

  render () {
    const { count } = this.props
    const boundClick = this.increment.bind(this, 'clicking')

    return (
      <Wrapper>
        <h1>Count: {count}</h1>
        <p>Click the button to increment the counter</p>
        <Button onClick={boundClick}>Increment</Button>
      </Wrapper>
    )
  }
}

export default Counter
