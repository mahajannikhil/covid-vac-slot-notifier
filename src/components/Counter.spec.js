import React from 'react'
import { mount } from 'enzyme'
import { matcher, serializer } from 'jest-glamor-react'
import { ThemeProvider } from 'glamorous'
import Counter from './Counter'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<Counter />', () => {
  const onIncrement = jest.fn()
  it('renders correctly', () => {
    const wrapper = mount(
      <ThemeProvider theme={{ colors: { green: '' } }}>
        <Counter onIncrement={onIncrement} />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshotWithGlamor()
    wrapper.find('button').simulate('click')
    expect(onIncrement).toHaveBeenCalled()
  })
})
