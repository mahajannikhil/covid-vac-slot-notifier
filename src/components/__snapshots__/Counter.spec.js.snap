// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
.css-abd50l,
[data-css-abd50l] {
  color: ;
}

.css-1fmls8,
[data-css-1fmls8] {
  border: none;
  color: white;
  padding: 1em;
  border-radius: 3px;
  cursor: pointer;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "green": "",
      },
    }
  }
>
  <Counter
    onIncrement={[Function]}
  >
    <glamorous(div)>
      <div
        className="css-abd50l"
      >
        <h1>
          Count: 
        </h1>
        <p>
          Click the button to increment the counter
        </p>
        <glamorous(button)
          onClick={[Function]}
        >
          <button
            className="css-1fmls8"
            onClick={[Function]}
          >
            Increment
          </button>
        </glamorous(button)>
      </div>
    </glamorous(div)>
  </Counter>
</ThemeProvider>
`;
